### PR TITLE
Dp 6426 pano viewer

### PR DESCRIPTION
**Pull request by aditudorache**

### CODE REVIEW

File: src/Hotspot.scss [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  outline: none;
+  padding: 0 20px;
+
+  &:hover {
+    animation: hotspot-animation 100ms linear;
+    transform: scale(1.15);
+  }
+}
+
+.c-hotspot--year-2016 {
```
Comment (chrisvanmook) [inline src/Hotspot.scss]: Rather a personal opinion: why repeat yourself while you can do &--year-2016 (the power of BEM and sass!)

File: src/services/angle-conversion/angle-conversion.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const ratio = Math.PI / 180;
+
+export const degreesToRadians = (degrees) => degrees * ratio;
```
Comment (chrisvanmook) [inline src/services/angle-conversion/angle-conversion.js:3]: Shouldn't we return the values with .toFixed(2)?

File: src/services/geo-json/geo-json.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+/* This thing has support for Point, Polygon en MultiPolygon. The thing is CRS agnostic.
+ *
+ * @param {Object} geoJson
+ *
+ * @returns {Object.<string, number>} A location in latitude and longitude
+ */
+export default function getCenter(geoJson) {
+  if (geoJson.type === 'Point') {
```
Comment (chrisvanmook) [inline src/services/geo-json/geo-json.js:31]: this is a bit odd: here you expect an array with 2 values, while you check the length of the coordinates in the getCoordinates method.. shouldn't we check this beforehand?

File: src/services/geo-json/geo-json.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  const isCoordinate = coordinates.length === 2 &&
+    typeof coordinates[0] === 'number' &&
+    typeof coordinates[1] === 'number';
+
+  if (isCoordinate) {
+    return [[coordinates[0]], [coordinates[1]]];
+  }
+
+  // We have to go deeper recursively; two levels for Polygons, three levels for MultiPolygons
+  return coordinates
```
Comment (chrisvanmook) [inline src/services/geo-json/geo-json.js:11]: what if we get a different param (other than polygons or multipolygons)? Throw an error? Catch it somewhere?

File: test/panoview.css [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  padding: 0;
+}
+
+body {
+  background-color: white;
+  font-family: sans-serif;
+  font-size: 10pt;
+}
+
+#selector {
```
Comment (chrisvanmook) [inline test/panoview.css:13]: ?

File: test/panoview.css [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  cursor: pointer;
+}
+
+.text {
+  margin-top: 5%;
+  font-size: 1.8rem;
+  font-weight: 300;
+}
+
+#info {
```
Comment (chrisvanmook) [inline test/panoview.css:43]: Why are we using id's as selectors?

File: webpack.prod.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
+const path = require('path');
```
Comment (chrisvanmook) [inline webpack.prod.js:1]: Please use a webpack.base to avoid repeating yourself

File: src/viewer.js [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+   * @param {float} [pitch] - initial pitch
+   * @param {float} [fov] - Field of vision
+   *
+   * This is the main api interaction for loading the panorama view.
+   * After this has been called navigating within the panorama via hotspots
+   * is handled by _updatePanorama
+   */
+  loadPanorama(lat, lon, tags, yaw, pitch, fov) {
+    // Updating POV if needed
+    this.pov.fov = fov || this.pov.fov;
```
Comment (chrisvanmook) [inline src/viewer.js]: Please rewrite this to use parameter defaults

File: src/utils.js [hunk 9]
```diff
@@ -1,71 +1,34 @@
```
Comment (chrisvanmook) [inline src/utils.js]: Unit test?

File: src/viewer.js [hunk 10]
```diff
@@ -0,0 +1,135 @@
+import { getElement } from './utils';
```
Comment (chrisvanmook) [inline src/viewer.js]: Unit test?

File: src/services/marzipano/marzipano.js [hunk 11]
```diff
@@ -0,0 +1,144 @@
+import Marzipano from 'marzipano';
```
Comment (chrisvanmook) [inline src/services/marzipano/marzipano.js:1]: Could we split this file into smaller bits? Perhaps make it a bit more reusable, it's doing a LOT of things here 😛  Also noticed a Todo, some eslint-disables and some arbitrary values in createHotspotTemplate. Also the addEventListener isn't being removed anywhere... Potential memory leak?

**Review by chrisvanmook — CHANGES_REQUESTED** [review]

File: src/Hotspot.scss (hunk 1 above)
Comment (aditudorache) [reply src/Hotspot.scss]: out of scope

File: src/services/angle-conversion/angle-conversion.js (hunk 2 above)
Comment (aditudorache) [reply src/services/angle-conversion/angle-conversion.js:3]: out of scope

File: src/services/geo-json/geo-json.js (hunk 3 above)
Comment (aditudorache) [reply src/services/geo-json/geo-json.js:31]: out of scope

File: src/services/geo-json/geo-json.js (hunk 4 above)
Comment (aditudorache) [reply src/services/geo-json/geo-json.js:11]: out of scope

File: src/viewer.js (hunk 8 above)
Comment (aditudorache) [reply src/viewer.js]: done

File: src/services/marzipano/marzipano.js (hunk 11 above)
Comment (aditudorache) [reply src/services/marzipano/marzipano.js:1]: out of scope

**Review by chrisvanmook — APPROVED** [review]

**Pull request merged**